addons/website_sale/models/sale_order.py

```diff
@@ -112,6 +112,15 @@ def _compute_payment_term_id(self):
             if order.website_id:
                 order.payment_term_id = order.website_id.with_company(order.company_id).sale_get_payment_term(order.partner_id)
 
+    def _compute_pricelist_id(self):
+        # Override to compute pricelists for carts using the partner's GeoIP
```

Comment:
Maybe slightly improve the comment to clarify that it's only a fallback when the partner has no address specified ?